File: src/prototype/ShapeRegistry.java
package prototype;

import java.util.HashMap;
import java.util.Map;

public class ShapeRegistry {
    private static Map<String, ShapePrototype> prototypes = new HashMap<>();

    static {
        // Prototipos iniciales
        prototypes.put("basic_circle", new Circle(10));
        prototypes.put("basic_rectangle", new Rectangle(20, 15));
    }

    public static ShapePrototype getShape(String type) {
        return prototypes.get(type).clone();
    }

    public static void addPrototype(String key, ShapePrototype prototype) {
        prototypes.put(key, prototype);
    }
}

File: Python/python/Prototype/ShapeRegistry.py
from .Circle import Circle
from .Rectangle import Rectangle
from .ShapePrototype import ShapePrototype

# 3. Registro de prototipos
class ShapeRegistry:
    _prototypes = {
        "basic_circle": Circle(10),
        "basic_rectangle": Rectangle(20, 15)
    }

    @classmethod
    def get_shape(cls, shape_name: str) -> ShapePrototype:
        prototype = cls._prototypes.get(shape_name)
        if prototype:
            return prototype.clone()
        raise ValueError(f"Prototipo no registrado: {shape_name}")

    @classmethod
    def add_prototype(cls, name: str, prototype: ShapePrototype):
        cls._prototypes[name] = prototype